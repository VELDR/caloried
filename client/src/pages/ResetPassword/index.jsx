import PropTypes from 'prop-types';
import { useDispatch } from 'react-redux';
import { Key } from '@mui/icons-material';
import { useState } from 'react';
import { useNavigate, useParams } from 'react-router-dom';
import { FormattedMessage, injectIntl } from 'react-intl';
import { jwtDecode } from 'jwt-decode';

import { resetPassword } from '@containers/Client/actions';
import FormInput from '@components/ui/FormInput';

import classes from './style.module.scss';

const ResetPassword = ({ intl: { formatMessage } }) => {
  const { token } = useParams();
  const dispatch = useDispatch();
  const navigate = useNavigate();
  const [password, setPassword] = useState('');

  const decoded = jwtDecode(token);

  const { email } = decoded;

  const handleReset = () => {
    dispatch(
      resetPassword({ email, newPassword: password, token }, () => {
        navigate('/sign-in');
      })
    );
  };

  return (
    <div className={classes.page}>
      <div className={classes.container}>
        <div className={classes.header}>
          <div className={classes.header__title}>
            <FormattedMessage id="app_reset_password_title" />
          </div>
          <div className={classes.header__description}>
            <FormattedMessage id="app_reset_password_description" />
          </div>
        </div>
        <div className={classes.form}>
          <FormInput
            label={formatMessage({ id: 'app_new_password' })}
            icon={<Key />}
            type="password"
            name="newPassword"
            onChange={(e) => setPassword(e.target.value)}
          />
          <div className={classes.form__button} onClick={handleReset}>
            <FormattedMessage id="app_reset" />
          </div>
        </div>
      </div>
    </div>
  );
};

ResetPassword.propTypes = {
  intl: PropTypes.object,
};

export default injectIntl(ResetPassword);
