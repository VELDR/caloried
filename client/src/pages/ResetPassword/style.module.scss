@import '@styles/base';

.page {
  @include flex-center;
  min-height: 80vh;
  padding: 1rem;
  .container {
    @include flex-center-column(1rem);
    background-color: hsla(var(--hue), var(--saturation), var(--lightness), 0.65);
    border-radius: 2rem;
    border: 1px solid var(--color-text-secondary);
    box-sizing: border-box;
    padding: 2rem;
    position: relative;
    text-align: center;
    .header {
      @include flex-center-column;
      width: fit-content;
      &__title {
        font-size: $font-size-larger;
      }
      &__description {
        width: 75%;
        font-size: $font-size-smaller;
      }
    }
    .form {
      @include flex-column(0.5rem);
      &__button {
        @include primary-button;
        font-size: $font-size-regular;
      }
    }
  }
}
