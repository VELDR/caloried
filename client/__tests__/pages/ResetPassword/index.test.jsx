import ResetPassword from '@pages/ResetPassword';
import { fireEvent, render } from '@utils/testHelper';

const mockNavigate = jest.fn();
const mockParams = {
  token:
    'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJlbWFpbCI6InRoZWdhbWluZ2Jhbm5lZEBnbWFpbC5jb20iLCJpYXQiOjE3MDI0NzY1NzYsImV4cCI6MTcwMjQ3NjYzNn0.ACQHc8Y_e5y-Nl3-FhbH2ph_Ecr78VwFA4NY8oZKk80',
};

jest.mock('react-router-dom', () => ({
  useNavigate: () => mockNavigate,
  useParams: () => mockParams,
}));

let wrapper;

beforeEach(() => {
  wrapper = render(<ResetPassword />);
});

describe('ResetPassword Page', () => {
  it('should render the page correctly', () => {
    const { getByTestId } = wrapper;

    expect(getByTestId('reset-password')).toBeInTheDocument();
  });

  it('should update password state on input change', () => {
    const { getByTestId } = wrapper;

    const passwordInput = getByTestId('form-input-password');

    fireEvent.change(passwordInput, { target: { value: 'newPassword123' } });

    expect(passwordInput.value).toBe('newPassword123');
  });

  it('should match with snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
