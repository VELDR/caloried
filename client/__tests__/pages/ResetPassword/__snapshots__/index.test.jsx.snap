// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResetPassword Page should match with snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="page"
        data-testid="reset-password"
      >
        <div
          class="container"
        >
          <div
            class="header"
          >
            <div
              class="header__title"
            >
              Atur Ulang Kata Sandi Anda
            </div>
            <div
              class="header__description"
            >
              Gunakan kata sandi yang aman dan mudah diingat
            </div>
          </div>
          <div
            class="form"
          >
            <div
              class="form"
            >
              <div
                class="form__label"
              >
                Sandi baru
              </div>
              <div
                class="form__input"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="KeyIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M21 10h-8.35C11.83 7.67 9.61 6 7 6c-3.31 0-6 2.69-6 6s2.69 6 6 6c2.61 0 4.83-1.67 5.65-4H13l2 2 2-2 2 2 4-4.04L21 10zM7 15c-1.65 0-3-1.35-3-3s1.35-3 3-3 3 1.35 3 3-1.35 3-3 3z"
                  />
                </svg>
                <input
                  data-testid="form-input-password"
                  name="newPassword"
                  type="password"
                  value=""
                />
                <div
                  class="eyeIcon"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                    data-testid="VisibilityIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M12 4.5C7 4.5 2.73 7.61 1 12c1.73 4.39 6 7.5 11 7.5s9.27-3.11 11-7.5c-1.73-4.39-6-7.5-11-7.5zM12 17c-2.76 0-5-2.24-5-5s2.24-5 5-5 5 2.24 5 5-2.24 5-5 5zm0-8c-1.66 0-3 1.34-3 3s1.34 3 3 3 3-1.34 3-3-1.34-3-3-3z"
                    />
                  </svg>
                </div>
              </div>
            </div>
            <div
              class="form__button"
            >
              Atur Ulang
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="page"
      data-testid="reset-password"
    >
      <div
        class="container"
      >
        <div
          class="header"
        >
          <div
            class="header__title"
          >
            Atur Ulang Kata Sandi Anda
          </div>
          <div
            class="header__description"
          >
            Gunakan kata sandi yang aman dan mudah diingat
          </div>
        </div>
        <div
          class="form"
        >
          <div
            class="form"
          >
            <div
              class="form__label"
            >
              Sandi baru
            </div>
            <div
              class="form__input"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="KeyIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M21 10h-8.35C11.83 7.67 9.61 6 7 6c-3.31 0-6 2.69-6 6s2.69 6 6 6c2.61 0 4.83-1.67 5.65-4H13l2 2 2-2 2 2 4-4.04L21 10zM7 15c-1.65 0-3-1.35-3-3s1.35-3 3-3 3 1.35 3 3-1.35 3-3 3z"
                />
              </svg>
              <input
                data-testid="form-input-password"
                name="newPassword"
                type="password"
                value=""
              />
              <div
                class="eyeIcon"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="VisibilityIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 4.5C7 4.5 2.73 7.61 1 12c1.73 4.39 6 7.5 11 7.5s9.27-3.11 11-7.5c-1.73-4.39-6-7.5-11-7.5zM12 17c-2.76 0-5-2.24-5-5s2.24-5 5-5 5 2.24 5 5-2.24 5-5 5zm0-8c-1.66 0-3 1.34-3 3s1.34 3 3 3 3-1.34 3-3-1.34-3-3-3z"
                  />
                </svg>
              </div>
            </div>
          </div>
          <div
            class="form__button"
          >
            Atur Ulang
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
